docs/contributing/upgrading-dependencies.md

```diff
@@ -3,6 +3,22 @@
 This document outlines how we manage the dependencies that GitHub Desktop needs,
 as some need to be handled differently to others
 
+## General Guidelines
+
+In the interest of stability and caution we tend to stay a version (or more) behind on our most critical dependencies. We vary whether this is a major, minor, or patch version depending on the development practices of the dependency's project.
+
+| Dependency  | Versions Behind Latest |
+| --- | --- |
+| electron | >= 1 major |
+| electron-builder | >= 1 major |
+| electron-packager | >= 1 major |
+| electron-winstaller | >= 1 major |
```

Comment:
I'm also not sure this one bumps it's major version often enough to constrain it like this:

 - we're on `2.5.2` which was tagged `Jan 26, 2017`
 - latest is `3.0.4` which was published `March 2 2019`

```suggestion
| electron-winstaller | >= 1 minor |
```